sdk/python/kfp/compiler/_op_to_template.py

```diff
@@ -138,6 +139,8 @@ def _inputs_to_json(
     artifacts = []
     for name, path in (input_artifact_paths or {}).items():
         artifact = {'name': name, 'path': path}
+        if name in artifact_arguments: # The arguments should be compiled as DAG task arguments, not template's default values, but in the current DSL-compiler implementation it's too hard to make that work when passing artifact references.
+            artifact['raw'] = {'data': str(artifact_arguments[name])}
```

Comment:
Sorry I'm not familiar with this. Are all artifacts guaranteed to be serializable?

Reply:
Good idea for the next PR.
Right now we only support two types of artifact arguments - constant string and `PipelineParam` reference to some task's output, which is also serializable.
In future I can put support for serializing complex types here so that it's on par with the component argument serialization support.